### ISSUE TITLE
narrowing conversion warnings

### ISSUE THREAD

**Issue by valassi** [madgraph5/madgraph4gpu#392]
Just as a reminder, seen this in float builds (icx, also others?)
Details later

**Comment by valassi** [comment]
Warnings like:

```
ccache /usr/local/cuda-11.6/bin/nvcc  -O3  -lineinfo -I. -I../../src -I../../../../../tools -I/usr/local/cuda-11.6/include/ -DUSE_NVTX -gencode arch=compute_70,code=compute_70 -gencode arch=compute_70,code=sm_70 -use_fast_math -std=c++17  -ccbin /cvmfs/projects.cern.ch/intelsw/oneAPI/linux/x86_64/2022/compiler/2022.0.2/linux/bin/icpx -allow-unsupported-compiler -DMGONGPU_FPTYPE_DOUBLE -DMGONGPU_HARDCODE_CIPC -Xcompiler -fPIC -c gRamboSamplingKernels.cu -o build.512y_d_inl0_hrd1/gRamboSamplingKernels.o
../../src/Parameters_sm.h:82:26: warning: comparison with infinity always evaluates to false in fast floating point modes [-Wtautological-constant-compare]
return ((x >= (0)) && (x < std::numeric_limits< double> ::infinity())) ? detailSqrtNewtonRaphson(x, x, 0) : std::numeric_limits< double> ::quiet_NaN(); 
                       ~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
In file included from Parameters_sm.cc:8:
./Parameters_sm.h:82:24: warning: comparison with infinity always evaluates to false in fast floating point modes [-Wtautological-constant-compare]
    return x >= 0 && x < std::numeric_limits<double>::infinity()
                     ~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

or

```
gCPPProcess.cu(50): warning #2361-D: invalid narrowing conversion from "double" to "float"
```

**Comment by valassi** [comment]
This is fixed in PR #393, which I am about to merge. CLosing.
